cypress-io/cypress
# chore: use `resolve-package-path` to resolve installed dependency versions more reliably

- Closes #25782

### Additional details


There are some dependencies that we can't resolve with [`require.resolve`](https://nodejs.org/api/modules.html#requireresolverequest-options), because of one of the following reasons:
- The package has `exports` defined, but `package.json` is not one of the exports
- Some packages don't have entrypoints (like in the case of [`react-scripts`](https://github.com/facebook/create-react-app/tree/main/packages/react-scripts), there is no entrypoint, just a bin).

We need to find a way around these limitations so that we can figure out if a dependency is installed, and if it is, which version is installed. Lachlan and I found a package called [`resolve-package-path`](https://www.npmjs.com/package/resolve-package-path) that does the trick, working around the `exports` and entrypoint limitations and allowing us to detect dependencies that are installed.

TLDR: It looks like we can't use `require.resolve` to figure out which dependencies of which packages are installed.

[Further reading](https://stackoverflow.com/questions/10111163/how-can-i-get-the-path-of-a-module-i-have-loaded-via-require-that-is-not-mine)

### Steps to test


1. `yarn create vite my-app --template react`
2. Install dependencies in `my-app`
3. cd into `node_modules/react`, remove `package.json` from the exports key in `package.json`
4. Open Launchpad with `my-app` and click Component Testing
5. Verify that the framework and bundler (React and Vite) are automatically detected and are selected in the dropdowns

### How has the user experience changed?


There should be no noticeable change for the user

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
